File: EXAMPORTALQUESTIONPAPERPOJO/ExamPortalApplication/src/main/java/com/app/service/QuestionService.java
package com.app.service;

import java.util.List;
import java.util.Set;

import com.app.entities.Question;
import com.app.entities.QuestionPaper;


public interface QuestionService {
	
	public void addQuestion(Long subject_id, Question question);
	
	public List<Question> getAllQuestionBySubject(Long subject_id);
	
	List<Question> limitedQuestion(Long subjectId);
	
	public Question getQuestion(Long question_id);
	
	public Set<Question> getQuestionOfQuestionPaper(QuestionPaper questionPaper);
	
	public void deleteQuestion(Long question_id);

}

File: MANUALPAPERGENRATION/ReactAppQuestionGeneration/myapp/src/service/question_service.js
import httpClient from "../http-common";

const getAllQuestionBySubject = (subject_id) => {
  return httpClient.get(`/course/subject/question/${subject_id}`);
};

const addQuestion = (question, subject_id) => {
  return httpClient.post(`/course/subject/question/${subject_id}`, question);
};

const limitedQuestion = (subject_id) => {
  return httpClient.get(`/course/subject/question/limit/${subject_id}`);
};

export default { getAllQuestionBySubject, addQuestion, limitedQuestion };
